namespace.go

```diff
@@ -120,6 +125,23 @@ func NamespaceWithWebSocket() NamespaceOption {
 	}
 }
 
+// NamespaceFromMSI configures a namespace from the environment (variables or MSI)
+func NamespaceFromMSI(name string) NamespaceOption {
```

Comment:
Hmm... This does a bit more than create a namespace with Managed Identity support (Managed Service Identity is the old name for Managed Identity --- at least that's what guidance I've received). This method will configure the namespace to use service principal from env or managed identity if available. I'm not sure this name reflects the true nature of the option.

Also, please prefix the name with `NamespaceWith...`.

I'm struggling to come up with the proper name right now. What do you think about `NamespaceWithEnvironmentBinding`? With that name, it calls to what it really does. The documentation for the method could then specify the cascade of functionality (SP w/ password, w/ certificate and also, managed identity).

Would it be better to split up the `JWTProvider` configuration to allow for more granular configuration, so you could only choose managed identity rather than the environmental cascade?

Reply:
@devigned `NamespaceWithEnvironmentBinding` sounds good actually. It does indicate accurately what the function does. I'll update the Docstring and the method's name.

Reply:
@devigned I've renamed the method and added a Docstring now.